applications/services/gui/modules/number_input.c

```diff
@@ -0,0 +1,493 @@
+#include "number_input.h"
+
+#include <gui/elements.h>
+#include <furi.h>
+#include <assets_icons.h>
+
+/** NumberInput type */
+struct NumberInput {
+    View* view;
+};
+
+typedef struct {
+    const char text;
+    const uint8_t x;
+    const uint8_t y;
+} NumberInputKey;
+
+typedef struct {
+    const char* header;
+    FuriString* text_buffer;
```

Comment:
Do you need text? At the end you are working with the numbers

Reply:
Yes, the input is a keyboard and therefore it creates a string, which is also shown on screen. 
When I follow your change requests to return a number in callback, I'd convert this string and remove it when the module is closed.